sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala

```diff
@@ -331,7 +331,7 @@ private[sql] class JDBCRDD(
       case BooleanType => BooleanConversion
       case DateType => DateConversion
       case DecimalType.Unlimited => DecimalConversion(None)
-      case DecimalType.Fixed(d) => DecimalConversion(Some(d))
+      case DecimalType.Fixed(d, s) => DecimalConversion(Some((d, s)))
```

Comment:
As said it was only about a warning, not about correctness. I'll drop this change on the next version, it draws too much attention and is not needed.